### PR TITLE
samples: ps_server: define GPIOs for "ot diag gpio"

### PR DESCRIPTION
**Pull request by Damian-Nordic**
Define "openthread_config" DTS node with the definition of GPIOs to be used by the "ot diag gpio" commands.

Btw, remove a leftover file for nRF54L15 PDK.

### CODE REVIEW

**Review by alxelax — COMMENTED** [review]
Hm... I do not think we need to assign double purpose for uart pins. It will break serial communication.
We need to choose another uart or assign another pins

File: samples/nrf_rpc/protocols_serialization/server/snippets/openthread/boards/nrf54l15dk_nrf54l15_cpuapp.overlay [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/* Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/ {
+	openthread_config: openthread {
+		compatible = "openthread,config";
+		diag-gpios = <&gpio2 9 GPIO_ACTIVE_HIGH>,		    // LED 0
+			     <&gpio1 10 GPIO_ACTIVE_HIGH>,		    // LED 1
```
[the remark this answers is not in the document]
Comment (Damian-Nordic) [reply samples/nrf_rpc/protocols_serialization/server/snippets/openthread/boards/nrf54l15dk_nrf54l15_cpuapp.overlay:10]: Yes, this is what I mentioned in the chat. This needs to be cleaned up. But it's not this PR that is wrong - it is well aligned with the definition of buttons and LEDs for this DK.

[the remark this answers is not in the document]
Comment (alxelax) [reply samples/nrf_rpc/protocols_serialization/server/snippets/openthread/boards/nrf54l15dk_nrf54l15_cpuapp.overlay:10]: I'm pretty sure it adds the ability of misconfiguration. How customer can understand that the adding button or LED will break serial communication?
Looks like hidden knowledge.

**Review by alxelax — APPROVED** [review]
I approve it, but it shall not be backported into 2.8.0

**Pull request closed without merging**